## Условие 1

### Скачать данные по ссылке https://www.kaggle.com/datasets/esratmaria/house-price-dataset-with-other-information Считать данные с помощью pandas Вывести на экран первые 5 строк Посмотреть на описание признаков и на их содержание

In [1]:
import pandas as pd
import numpy as np

In [9]:
df = pd.read_csv('./kc_house_data.csv')
df.head()

id             date     price  bedrooms  bathrooms  sqft_living  \
0  7129300520  20141013T000000  221900.0         3       1.00         1180   
1  6414100192  20141209T000000  538000.0         3       2.25         2570   
2  5631500400  20150225T000000  180000.0         2       1.00          770   
3  2487200875  20141209T000000  604000.0         4       3.00         1960   
4  1954400510  20150218T000000  510000.0         3       2.00         1680   

   sqft_lot  floors  waterfront  view  ...  grade  sqft_above  sqft_basement  \
0      5650     1.0           0     0  ...      7        1180              0   
1      7242     2.0           0     0  ...      7        2170            400   
2     10000     1.0           0     0  ...      6         770              0   
3      5000     1.0           0     0  ...      7        1050            910   
4      8080     1.0           0     0  ...      8        1680              0   

   yr_built  yr_renovated  zipcode      lat     long  sqft_living15  \
0      1955             0    98178  47.5112 -122.257           1340   
1      1951          1991    98125  47.7210 -122.319           1690   
2      1933             0    98028  47.7379 -122.233           2720   
3      1965             0    98136  47.5208 -122.393           1360   
4      1987             0    98074  47.6168 -122.045           1800   

   sqft_lot15  
0        5650  
1        7639  
2        8062  
3        5000  
4        7503  

[5 rows x 21 columns]

## Условие 2

### Проведите первичный анализ данных Изучите типы данных Найдите количество пропущенных ячеек в данных Посчитайте основные статистики по всем признакам и поизучайте их Пишите выводы

In [10]:
df.shape

(21613, 21)

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21613 entries, 0 to 21612
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   id             21613 non-null  int64  
 1   date           21613 non-null  object 
 2   price          21613 non-null  float64
 3   bedrooms       21613 non-null  int64  
 4   bathrooms      21613 non-null  float64
 5   sqft_living    21613 non-null  int64  
 6   sqft_lot       21613 non-null  int64  
 7   floors         21613 non-null  float64
 8   waterfront     21613 non-null  int64  
 9   view           21613 non-null  int64  
 10  condition      21613 non-null  int64  
 11  grade          21613 non-null  int64  
 12  sqft_above     21613 non-null  int64  
 13  sqft_basement  21613 non-null  int64  
 14  yr_built       21613 non-null  int64  
 15  yr_renovated   21613 non-null  int64  
 16  zipcode        21613 non-null  int64  
 17  lat            21613 non-null  float64
 18  long  

In [12]:
df.describe()

id         price      bedrooms     bathrooms   sqft_living  \
count  2.161300e+04  2.161300e+04  21613.000000  21613.000000  21613.000000   
mean   4.580302e+09  5.400881e+05      3.370842      2.114757   2079.899736   
std    2.876566e+09  3.671272e+05      0.930062      0.770163    918.440897   
min    1.000102e+06  7.500000e+04      0.000000      0.000000    290.000000   
25%    2.123049e+09  3.219500e+05      3.000000      1.750000   1427.000000   
50%    3.904930e+09  4.500000e+05      3.000000      2.250000   1910.000000   
75%    7.308900e+09  6.450000e+05      4.000000      2.500000   2550.000000   
max    9.900000e+09  7.700000e+06     33.000000      8.000000  13540.000000   

           sqft_lot        floors    waterfront          view     condition  \
count  2.161300e+04  21613.000000  21613.000000  21613.000000  21613.000000   
mean   1.510697e+04      1.494309      0.007542      0.234303      3.409430   
std    4.142051e+04      0.539989      0.086517      0.766318      0.650743   
min    5.200000e+02      1.000000      0.000000      0.000000      1.000000   
25%    5.040000e+03      1.000000      0.000000      0.000000      3.000000   
50%    7.618000e+03      1.500000      0.000000      0.000000      3.000000   
75%    1.068800e+04      2.000000      0.000000      0.000000      4.000000   
max    1.651359e+06      3.500000      1.000000      4.000000      5.000000   

              grade    sqft_above  sqft_basement      yr_built  yr_renovated  \
count  21613.000000  21613.000000   21613.000000  21613.000000  21613.000000   
mean       7.656873   1788.390691     291.509045   1971.005136     84.402258   
std        1.175459    828.090978     442.575043     29.373411    401.679240   
min        1.000000    290.000000       0.000000   1900.000000      0.000000   
25%        7.000000   1190.000000       0.000000   1951.000000      0.000000   
50%        7.000000   1560.000000       0.000000   1975.000000      0.000000   
75%        8.000000   2210.000000     560.000000   1997.000000      0.000000   
max       13.000000   9410.000000    4820.000000   2015.000000   2015.000000   

            zipcode           lat          long  sqft_living15     sqft_lot15  
count  21613.000000  21613.000000  21613.000000   21613.000000   21613.000000  
mean   98077.939805     47.560053   -122.213896    1986.552492   12768.455652  
std       53.505026      0.138564      0.140828     685.391304   27304.179631  
min    98001.000000     47.155900   -122.519000     399.000000     651.000000  
25%    98033.000000     47.471000   -122.328000    1490.000000    5100.000000  
50%    98065.000000     47.571800   -122.230000    1840.000000    7620.000000  
75%    98118.000000     47.678000   -122.125000    2360.000000   10083.000000  
max    98199.000000     47.777600   -121.315000    6210.000000  871200.000000

In [13]:
df.describe(include=['object'])

date
count             21613
unique              372
top     20140623T000000
freq                142

### пропущенных ячеек нету.  по дате. уникальных 372. бальше всего встречаеться дата 23.06.2014(142 раза). средняя цена 450000. среднее колл. спальных комнат 3

## Условие 3

### 3.1 В каком диапазоне изменяются стоимости недвижимости?

In [15]:
df['price'].min()

75000.0

In [16]:
df['price'].max()

7700000.0

### стоимость недвижимости от 75000 до 7700000

### 3.2 Какую долю в среднем занимают жилая площадь от всей площади по всем домам?

In [18]:
df[['sqft_living','sqft_lot']].mean()

sqft_living     2079.899736
sqft_lot       15106.967566
dtype: float64

In [19]:
sqft = 2079/(15106/100)
sqft

13.76274328081557

### 13%

### 3.3 Как много домов с разными этажами в данных?

In [20]:
df['floors'].value_counts()

1.0    10680
2.0     8241
1.5     1910
3.0      613
2.5      161
3.5        8
Name: floors, dtype: int64

### 3.4 Насколько хорошие состояния у домов в данных?

In [21]:
df['condition'].value_counts()

3    14031
4     5679
5     1701
2      172
1       30
Name: condition, dtype: int64

### 3.5 Найдите года, когда построили первый дом, когда построили последний дом в данных?

In [30]:
df['yr_built'].min()

1900

In [31]:
df['yr_built'].max()

2015

## Условие 4

### 4.1 Сколько в среднем стоят дома, у которых 2 спальни?

In [26]:
s = df[df['bedrooms'] == 2]
s[['price']].mean()

price    401372.681884
dtype: float64

### 4.2 Какая в среднем общая площадь домов, у которых стоимость больше 600 000?

In [28]:
s1 = df[df['price'] >= 600000]
s1[['sqft_lot']].mean()

sqft_lot    20454.783538
dtype: float64

### 4.3 Как много домов коснулся ремонт?

In [29]:
s2 = df[df['yr_renovated'] > 0]
s2

id             date      price  bedrooms  bathrooms  \
1      6414100192  20141209T000000   538000.0         3       2.25   
35     9547205180  20140613T000000   696000.0         3       2.50   
95     1483300570  20140908T000000   905000.0         4       2.50   
103    2450000295  20141007T000000  1088000.0         3       2.50   
115    3626039325  20141121T000000   740500.0         3       3.50   
...           ...              ...        ...       ...        ...   
19622  7351200295  20150114T000000  1150000.0         3       1.75   
20057   126039256  20140904T000000   434900.0         3       2.00   
20444  4305600360  20150225T000000   500012.0         4       2.50   
20447  3319500628  20150212T000000   356999.0         3       1.50   
20962  1278000210  20150311T000000   110000.0         2       1.00   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
1             2570      7242     2.0           0     0  ...      7   
35            2300      3060     1.5           0     0  ...      8   
95            3300     10250     1.0           0     0  ...      7   
103           2920      8113     2.0           0     0  ...      8   
115           4380      6350     2.0           0     0  ...      8   
...            ...       ...     ...         ...   ...  ...    ...   
19622         1760      6788     2.0           1     4  ...      7   
20057         1520      5040     2.0           0     0  ...      7   
20444         2400      9612     1.0           0     0  ...      8   
20447         1010      1546     2.0           0     0  ...      8   
20962          828      4524     1.0           0     0  ...      6   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
1            2170            400      1951          1991    98125  47.7210   
35           1510            790      1930          2002    98115  47.6827   
95           2390            910      1946          1991    98040  47.5873   
103          2920              0      1950          2010    98004  47.5814   
115          2780           1600      1900          1999    98117  47.6981   
...           ...            ...       ...           ...      ...      ...   
19622        1760              0      1940          1960    98125  47.7336   
20057        1520              0      1977          2006    98177  47.7770   
20444        1230           1170      1962          2009    98059  47.4799   
20447        1010              0      1971          2014    98144  47.5998   
20962         828              0      1968          2007    98001  47.2655   

          long  sqft_living15  sqft_lot15  
1     -122.319           1690        7639  
35    -122.310           1590        3264  
95    -122.249           1950        6045  
103   -122.196           2370        8113  
115   -122.368           1830        6350  
...        ...            ...         ...  
19622 -122.284           1630        7588  
20057 -122.362           1860        8710  
20444 -122.127           2430        5539  
20447 -122.311           1010        1517  
20962 -122.244            828        5402  

[914 rows x 21 columns]

### 914 домов

### 4.4 Насколько в среднем стоимость домов с оценкой grade домов выше 10 отличается от стоимости домов с оценкой grade меньше 4?

In [35]:
s3 = df[df['grade'] > 10]
s3[['price']].mean()

price    1.678635e+06
dtype: float64

In [38]:
s4 = df[df['grade'] < 4]
s4[['price']].mean()

price    189750.0
dtype: float64

In [39]:
ot = 1678635 - 189740
ot

1488895

## Условие 5

### 5.1 Выберите дом клиенту. Клиент хочет дом с видом на набережную, как минимум с тремя ванными и с подвалом. Сколько вариантов есть у клиента?

In [41]:
s5 = df[(df['waterfront'] > 0) & (df['bathrooms'] >= 3) & (df['sqft_basement'] > 0)]
s5.shape

(41, 21)

### 41 вариант есть у клиента

### 5.2 Выберите дом клиенту. Клиент хочет дом либо с очень красивым видом из окна, либо с видом на набережную, в очень хорошем состоянии и год постройки не меньше 1980 года. В какой ценовом диапазоне будут дома?

In [44]:
s6 = df[((df['view'] > 0) | (df['waterfront'] > 0)) & (df['grade'] >= 10) & (df['yr_built'] >= 1980)]
s6[['price']].min()

price    380000.0
dtype: float64

In [45]:
s6[['price']].max()

price    6885000.0
dtype: float64

### ценовой диапозон от 380000 до 6885000

### 5.3 Выберите дом клиенту. Клиент хочет дом без подвала, с двумя этажами, стоимостью до 150000. Какая оценка по состоянию у таких домов в среднем?

In [48]:
s7 = df[(df['sqft_basement'] == 0) & (df['floors'] == 2) & (df['price'] < 150000)]
s7[['grade']].mean()

grade    7.0
dtype: float64